app/backend/api.ts

```diff
@@ -56,3 +57,22 @@ export const getPaymentDetails = async (payeeName: string, partPostcode: string)
 
 	return JSON.parse(response.payload)
 }
+
+export const getDownloadDetailsCsv = async (payeeName: string, partPostcode: string): Promise<Buffer> => {
+	const url = getUrlParams('downloaddetails', {
+		payeeName,
+		partPostcode
+	})
+	return getBufferFromUrl(url)
+}
+
+const getBufferFromUrl = async (url:string): Promise<Buffer> => {
+	try {
+		const response = await fetch(`${config.backendEndpoint}${url}`)
+		const arrayBuffer = await response.arrayBuffer();
+		return Buffer.from(arrayBuffer);
+	} catch (error) {
+		console.log("Error while reading from URL " + url , error)
+		throw error
```

Comment:
I don't think its a good idea to throw an error here unless you intend to catch and handle the error outside of this function. From what I can see it's not getting handled elsewhere and this cascades all the way to downloaddetails where it just leads to internal server error and the user is directed to blank internal server error page. I would suggest returning a default value back after the exception is thrown and handle it from the caller. Please take a look at the paymentdata handler and its subsequent call to api as an example.

Reply:
We agreed that we would like to add a generic error page for error code 500 in another ticket as its outside the scope of this ticket.